sdk/core/azure-core/src/main/java/com/azure/core/util/FluxUtil.java

```diff
@@ -420,6 +458,71 @@ private static Context toAzureContext(ContextView context) {
         return returnContext;
     }
 
+    /**
+     * Writes the {@link ByteBuffer ByteBuffers} emitted by a {@link Flux} of {@link ByteBuffer} to an {@link
+     * OutputStream}.
+     * <p>
+     * The {@code stream} is not closed by this call, closing of the {@code stream} is managed by the caller.
+     * <p>
+     * The response {@link Mono} will emit an error if {@code content} or {@code stream} are null. Additionally, an
+     * error will be emitted if an exception occurs while writing the {@code content} to the {@code stream}.
+     *
+     * @param content The {@link Flux} of {@link ByteBuffer} content.
+     * @param stream The {@link OutputStream} being written into.
+     * @return A {@link Mono} which emits a completion status once the {@link Flux} has been written to the {@link
+     * OutputStream}, or an error status if writing fails.
+     */
+    public static Mono<Void> writeToOutputStream(Flux<ByteBuffer> content, OutputStream stream) {
+        return Mono.create(emitter -> {
+            if (content == null && stream == null) {
+                emitter.error(new NullPointerException("'content' and 'stream' cannot be null."));
+                return;
+            } else if (content == null) {
+                emitter.error(new NullPointerException("'content' cannot be null."));
+                return;
+            } else if (stream == null) {
+                emitter.error(new NullPointerException("'stream' cannot be null."));
+                return;
+            }
```

Comment:
nit: would it be more efficient/readable to do these checks outside of `Mono.create()` and return `monoError()` (exception will be logged too)?

Reply:
Readable yeah, definitely, not sure if there'll be an efficiency gain though